core/embed/rust/src/trezorhal/fatal_error.rs

```diff
@@ -5,70 +5,79 @@ mod ffi {
     }
 }
 
-use crate::ui::ui_features::{ModelUI, UIFeaturesCommon};
+use crate::ui::{
+    shape,
+    ui_features::{ModelUI, UIFeaturesCommon},
+};
+
+#[cfg(feature = "debug")]
+use crate::dbg_println;
 
 fn shutdown() -> ! {
     unsafe { ffi::trezor_shutdown() }
 }
 
-#[cfg(feature = "bootloader")]
-pub fn __fatal_error(_expr: &str, _msg: &str, _file: &str, _line: u32, _func: &str) -> ! {
-    ModelUI::screen_fatal_error("BL.rs", "BL.rs", "PLEASE VISIT\nTREZOR.IO/RSOD");
+/// Shows an error message and shuts down the device.
+pub fn error_shutdown(title: &str, msg: &str, footer: &str) -> ! {
+    // SAFETY:
+    // The is the only situation we are allowed use this function
```

Comment:
typo This is

Reply:
fixed, see 640dee638a19a7ee831cfda8d92fb8cdafa5713b